pkg/api/validation/validation.go

```diff
@@ -71,13 +71,10 @@ func ValidateAnnotations(annotations map[string]string, field string) errs.Valid
 		if !util.IsQualifiedName(strings.ToLower(k)) {
 			allErrs = append(allErrs, errs.NewFieldInvalid(field, k, qualifiedNameErrorMsg))
 		}
-		if !util.IsValidAnnotationValue(v) {
-			allErrs = append(allErrs, errs.NewFieldInvalid(field, k, ""))
-		}
 		totalSize += (int64)(len(k)) + (int64)(len(v))
 	}
 	if totalSize > (int64)(totalAnnotationSizeLimitB) {
-		allErrs = append(allErrs, errs.NewFieldTooLong("annotations", ""))
+		allErrs = append(allErrs, errs.NewFieldTooLong(field, "", fmt.Sprintf("%d B", totalAnnotationSizeLimitB)))
```

Comment:
I looked again at this code and I've realized that FieldTooLong makes sense only for text fields (at least that's the current use-case). So I've just changed the argument to int and updated the message accordingly.